src/Archives/VolFile.cpp

```diff
@@ -282,11 +282,13 @@ namespace Archives
 			volInfo.indexEntries.push_back(indexEntry);
 
 			// Add length of internal filename plus null terminator to string table length.
-			volInfo.stringTableLength += volInfo.internalNames[i].size() + 1;
+			// stringTableLength cannot be greater in size than a 4 byte integer.
+			volInfo.stringTableLength += static_cast<uint32_t>(volInfo.internalNames[i].size()) + 1;
 		}
 
 		// Calculate size of index table
-		volInfo.indexTableLength = volInfo.fileCount() * sizeof(IndexEntry);
+		// indexTableLength cannot be greater in size than a 4 byte integer
+		volInfo.indexTableLength = static_cast<uint32_t>(volInfo.fileCount()) * sizeof(IndexEntry);
```

Comment:
Should be fixed in next set of commits.

Reply:
On 64-bit builds, this check could theoretically overflow for really large `volInfo` vectors. This isn't a hugely practical concern though, since any real-world system would run out of memory before reaching that point.